### PR TITLE
[Part1][Live-scale] Support hotreload and add skeleton for ServiceScaleManager

### PR DESCRIPTION
**Pull request by JialinXin**
1. Add `IMultiEndpointServiceConnectionContainer` top of `IServiceConnectionContainer` to support add/remove `HubServiceEndpoint`
2. Add `IMultiEndpointServiceContainerManager` to manager `IMultiEndpointServiceConnectionContainer`
3. Add `IServiceScaleManager` to support Add/Remove/Get ServiceEndpoint
4. Support hot-reload configurations in AspNetCore targeting Endpoints configuration changes.
5. Add `EnableAutoScale` in ServiceOptions as flag of hot-reload configuration of endpoints.

TODO:
1. Status sync with ASRS
2. AspNet support
3. Implement ServiceScaleManagerBase
4. UTs.

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Common/Constants.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@ internal static class Constants
         public const string ServerStickyModeDefaultKey = "Azure:SignalR:ServerStickyMode";
         public const string ConnectionStringDefaultKey = "Azure:SignalR:ConnectionString";
         public const string ApplicationNameDefaultKey = "Azure:SignalR:ApplicationName";
+        public const string EndpointsDefaultKey = "Azure:SignalR:Endpoints";
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Constants.cs]: >Endpoints [](start = 65, length = 9)

`ConnectionString` is reused for `Endpoints` already

**Comment by vicancy** [comment]
Can it be separated to smaller PRs?

File: src/Microsoft.Azure.SignalR.Common/Constants.cs (hunk 1 above)
Comment (JialinXin) [reply src/Microsoft.Azure.SignalR.Common/Constants.cs]: Yes, you're right. Didn't notice it worked for multiple endpoints. Will revert this.

**Comment by JialinXin** [comment]
OK. Will separate hot-reload support and interface changes to different PRs.

**Pull request closed without merging**